### ISSUE TITLE
Failed to read register DRW at address 0x0000000c

### ISSUE THREAD

**Issue by fominok** [rp-rs/rp2040-project-template#23]
Hello,
Please redirect me if this is the wrong resource to post this kind of issue, but I've found no place better than this issue tracker as I use this template to run Rust on my RPi Pico;

With the guide in the Readme I connected two Picos, making one of them a probe and run debug build like this:
```
DEFMT_LOG=trace cargo run
```

It seems working, but it lasts only for a few seconds:

```
     Running `probe-run --chip RP2040 target/thumbv6m-none-eabi/debug/rp2040-project-template`
(HOST) INFO  flashing program (4 pages / 16.00 KiB)
(HOST) INFO  success!
────────────────────────────────────────────────────────────────────────────────
INFO  Program start
└─ rp2040_project_template::__cortex_m_rt_main @ src/main.rs:28
INFO  on!
└─ rp2040_project_template::__cortex_m_rt_main @ src/main.rs:60
INFO  off!
└─ rp2040_project_template::__cortex_m_rt_main @ src/main.rs:63
INFO  on!
└─ rp2040_project_template::__cortex_m_rt_main @ src/main.rs:60
INFO  off!
└─ rp2040_project_template::__cortex_m_rt_main @ src/main.rs:63
INFO  on!
└─ rp2040_project_template::__cortex_m_rt_main @ src/main.rs:60
INFO  off!
└─ rp2040_project_template::__cortex_m_rt_main @ src/main.rs:63
INFO  on!
└─ rp2040_project_template::__cortex_m_rt_main @ src/main.rs:60
INFO  off!
└─ rp2040_project_template::__cortex_m_rt_main @ src/main.rs:63
INFO  on!
└─ rp2040_project_template::__cortex_m_rt_main @ src/main.rs:60
RTT error: Error communicating with probe: A core architecture specific error occured
────────────────────────────────────────────────────────────────────────────────
Error: A core architecture specific error occured

Caused by:
    0: Failed to read register DRW at address 0x0000000c
    1: An error specific to a probe type occured
    2: Error handling CMSIS-DAP command Transfer
    3: Timeout in USB communication.

```

If I re-run this command, it won't even start (and blinky stops work as well):

```
     Running `probe-run --chip RP2040 target/thumbv6m-none-eabi/debug/rp2040-project-template`
(HOST) INFO  flashing program (4 pages / 16.00 KiB)
Error: Error while flashing

Caused by:
    0: Something during the interaction with the core went wrong
    1: A core architecture specific error occured
    2: Failed to write register DRW at address 0x0000000c
    3: An error specific to a probe type occured
    4: Error handling CMSIS-DAP command TransferBlock
    5: Timeout in USB communication.
```

However, when I reconnect the target Pico, it flashes and runs for a sec as before, then the same issue starts again.

**Comment by 9names** [comment]
Does this happen with a --release build as well?
I think the most likely problem here is a power issue - try changing USB cables and which port you're plugging into to see if it still happens.
Also check your grounding between the two boards.

**Comment by fominok** [comment]
@9names thank you for your response

I tried to use another host (previously it was RPi4, now it is a x64 laptop) and it works fine now, no idea why I didn't try it before. Your guess about power issue in case of RPi makes even more sense, as this device may complain even if I connect a keyboard, so I tried new cables and active USB-hub, no luck, unfortunately.
Anyway, I'm glad it works at least on my laptop, but maybe you have any idea why it doesn't on RPi?
